### ISSUE TITLE
Trying to reproduce speech translation results on Must-C 

### ISSUE THREAD

**Issue by bhaddow** [pytorch/fairseq#3422]
#### What is your question?

Hi

I am trying to reproduce the speech translation results you give on Must-C, at the foot of https://github.com/pytorch/fairseq/blob/master/examples/speech_to_text/docs/mustc_example.md .

If I download the pre-trained models (thanks for releasing these!), I get the same bleu scores on the languages I have checked. But training my own models resulted in lower scores, so I compared my configuration against the pre-trained models. 

The first problem is that, if I use the command recommended in the documentation, it sets `label-smoothing` to 0.0 (the default). In the pre-trained models it is set to 0.1, and indeed setting `label-smoothing` to 0.1 improves bleu by 1-2 points - could the documentation be updated with this setting?

My best scores are still lower than the pre-trained models (26.5 vs 27.2 for en-es and 21.5 vs 22.6 for en-de). I noticed that the pre-trained models use `label_smoothed_cross_entropy_with_accuracy` as their loss function, but this is not available in the current fairseq (as far as I can see). So my question is, what is the `label_smoothed_cross_entropy_with_accuracy`  loss, and did it improve performance over  using `label_smoothed_cross_entropy` ?

best
Barry 

#### What's your environment?

 - fairseq Version: master
 - PyTorch Version: 1.8.1
 - OS: Linux
 - How you installed fairseq (`pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

**Comment by sugeeth14** [comment]
Hello @bhaddow I am not able to train in the first place and facing this issue https://github.com/pytorch/fairseq/issues/3457
Did you follow the same process to train your model ?

**Comment by bhaddow** [comment]
Hi @sugeeth14 - I followed the instructions in the documentation - I left a comment in #3457 which may help.

**Comment by sugeeth14** [comment]
Hi, thanks for your comment it has helped. @bhaddow

**Comment by muhdhuz** [comment]
I'm also not able to run inference on the provided pretrained ST model due to the use of  label_smoothed_cross_entropy_with_accuracy which I don't see in the current build of fairseq. But training from scratch with label_smoothed_cross_entropy then running inference works.

**Comment by bhaddow** [comment]
@muhdhuz - I had to add `--model-overrides '{ "criterion" : "label_smoothed_cross_entropy"}'` to `fairseq-generate` in order to translate with the pretrained models.

**Comment by holyma** [comment]
@bhaddow I have trained mustc ASR/ST and got terrible results #3897 , all as this [instructions](https://github.com/pytorch/fairseq/blob/fcca32258c/examples/speech_to_text/docs/mustc_example.md), I don't know where is the problem？

**Comment by bhaddow** [comment]
I do not know where the problem is with your experiments. My results were good, but consistently worse than the pre-trained models.

**Comment by holyma** [comment]
> I do not know where the problem is with your experiments. My results were good, but consistently worse than the pre-trained models.

How many num_steps/epochs and how are the results？

**Comment by holyma** [comment]
> I do not know where the problem is with your experiments. My results were good, but consistently worse than the pre-trained models.

when I train the S2T model, always show 'CUDA OOM' warning. So I try to reduce `max-tokens` , at the same time, Do I need to change some other params (such as `warmup-updates` and `max-update`)?

**Comment by bhaddow** [comment]
If you get OOM, then you should reduce the batch size (and maybe increase gradient accumulation to compensate)

**Comment by gegallego** [comment]
Hi, I was quite surprised about why I was getting always around 1 BLEU below the official results in MuST-C. So, I've been checking for updates in this issue for a while.

I've just discovered that I wasn't doing the checkpoint average. I don't know why my eyes were jumping those lines in the README! Doing it closed the gap between the official results and mine.

It may seem obvious, I know, it's just following the instructions… but I leave the comment here anyway, in case someone has forgotten about this step too!
